### ISSUE TITLE
Bot responds to PM even if message sent in group

### ISSUE THREAD

**Issue by YouTubeATP** [aaronetz/signal-bot#2]
**Describe the bug**
The bot responds in private messages even if command is sent in group

**To Reproduce**
1. Send message to bot in group
2. Bot responds in private message

**Expected behavior**
Bot sends response to group

**Comment by aaronetz** [comment]
My guess would be that this is related to groups v2. These didn't exist back when I implemented the bot. I'm not able to look into this right now but I'm happy to merge a fix if you have one! Otherwise I'll get to it at some point.

**Comment by YouTubeATP** [comment]
I would love to but my Java skills are just very basic.
Looking forward and thanks

**Comment by YouTubeATP** [comment]
Yes, this is related to groups v2 as the bot replies just fine when in a "legacy group".

**Comment by APratham** [comment]
@YouTubeATP Can you please tell what are legacy groups?

**Comment by YouTubeATP** [comment]
> @YouTubeATP Can you please tell what are legacy groups? 

They are basically the groups back when Signal was in a older version. These groups did not support group admins and @mentions and more, while the new ones do.
In order to create a legacy group nowadays there must be at least one member who hasn't updated Signal to a version that supports the new groups

**Comment by APratham** [comment]
Thank you so much for the help!

**Comment by Giszmo** [comment]
I'd need a signal bot for non-legacy-groups. I offer $100 US sent in BTC for whoever fixes this issue.

Independent of this offer, @aaronetz would you know if there was a better Signal bot around, without the quite severe limitations mentioned in the Readme.md (and found here)?

**Comment by YouTubeATP** [comment]
> I'd need a signal bot for non-legacy-groups. I offer $100 in BTC for whoever fixes this issue.
> 
> Independent of this offer, @aaronetz would you know if there was a better Signal bot around, without the quite severe limitations mentioned in the Readme.md (and found here)?

Bro that's 3,170,780.00 USD. Are you sure?

**Comment by Giszmo** [comment]
> > I'd need a signal bot for non-legacy-groups. I offer $100 in BTC for whoever fixes this issue.
> > Independent of this offer, @aaronetz would you know if there was a better Signal bot around, without the quite severe limitations mentioned in the Readme.md (and found here)?
> 
> Bro that's 3,170,780.00 USD. Are you sure?

$100 is $100. The $ sign is being used for many currencies, the USD being the highest valued one to my knowledge. I have not seen anyone using your interpretation of "$" but made it more clear above nevertheless :grinning:.

**Comment by YouTubeATP** [comment]
> > > I'd need a signal bot for non-legacy-groups. I offer $100 in BTC for whoever fixes this issue.
> > > Independent of this offer, @aaronetz would you know if there was a better Signal bot around, without the quite severe limitations mentioned in the Readme.md (and found here)?
> > 
> > Bro that's 3,170,780.00 USD. Are you sure?
> 
> $100 is $100. The $ sign is being used for many currencies, the USD being the highest valued one to my knowledge. I have not seen anyone using your interpretation of "$" but made it more clear above nevertheless :grinning:.

Well, I misunderstood your statement, guess I was not being careful.

**Comment by aaronetz** [comment]
Just to keep you posted, I brought the code base up to date with the latest signal library, but supporting new groups would be a significant amount of work (they are much more complex than legacy groups) and I presently don't have the time to work on this. But I would definitely welcome anyone who wants to give it a shot.

**Comment by aaronetz** [comment]
Anyone who wants to give it a shot, you can look at the excellent https://github.com/AsamK/signal-cli for reference. It's a command line interface for Signal and it has been updated to support the new Groups.
